monitoring/v1alpha1/types.go

```diff
@@ -0,0 +1,342 @@
+	// evaluated at the current time, and all resultant time series become pending
+	// or firing alerts.  This is most often a string representing a PromQL
+	// expression, e.g.:
+	//
+	//   mapi_current_pending_csr > mapi_max_pending_csr
+	//
+	// In rare cases this could be a simple integer, e.g. a simple "1" if the
+	// intent is to create an alert that is always firing.  This is sometimes used
+	// to create an always-firing "Watchdog" alert in order to ensure the alerting
+	// pipeline is functional.
+	//
+	// +required
+	Expr intstr.IntOrString `json:"expr"`
+
+	// for is the time period after which alerts are considered firing after first
+	// returning results.  Alerts which have not yet fired for long enough are
+	// considered pending. This is represented as a Prometheus duration, for
+	// details on the format see:
+	// - https://prometheus.io/docs/prometheus/latest/configuration/configuration/#duration
+	//
+	// +kubebuilder:validation:Pattern:="((([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?|0)"
+	// +optional
+	For string `json:"for,omitempty"`
+
+	// labels to add or overwrite for each alert.  The results of the PromQL
+	// expression for the alert will result in an existing set of labels for the
+	// alert, for any label specified here with the same name as a label in that
+	// set, the label here wins and overwrites the previous value.  These should
+	// typically be short identifying values that may be useful to query against.
+	// A common example is the alert severity:
+	//
+	//   labels:
+	//     severity: warning
+	//
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// annotations to add to each alert.  These are values that can be used to
+	// store longer additional information that you won't query on, such as alert
+	// descriptions or runbook links, e.g.:
+	//
+	//   annotations:
+	//     summary: HAProxy reload failure
+	//     description: |
+	//       This alert fires when HAProxy fails to reload its
+	//       configuration, which will result in the router not picking up
+	//       recently created or modified routes.
+	//
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// AlertingRuleStatus is the status of an AlertingRule resource.
+type AlertingRuleStatus struct {
+	// observedGeneration is the last generation change you've dealt with.
+	//
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// prometheusRule is the generated PrometheusRule for this AlertingRule.
+	//
+	// +optional
+	PrometheusRule PrometheusRuleRef `json:"prometheusRule,omitempty"`
+}
+
+// PrometheusRuleRef is a reference to an existing PrometheusRule object.  Each
+// AlertingRule instance results in a generated PrometheusRule object in the same
+// namespace, which is always the openshift-monitoring namespace.  This is used to
+// point to the generated PrometheusRule object in the AlertingRule status.
```

Comment:
This context won't end up in the generated docs, only the docs from where the type is referenced, perhaps we can copy these notes to the AlertinRuleStatus definition to make sure they end up in the documentation

Reply:
Done.